lib/cisco_node_utils/vxlan_vtep.rb

```diff
@@ -90,121 +90,48 @@ def description=(desc)
-    #      final_hash[match_arr[1]] = "#{match_arr[2]}"
-    #    end
-    #  end
-    #  final_hash
-    # end
-
-    # TODO: Move vni_mcast_grp_map to vxlan_vtp_vni.rb object
-    #
-    # def vni_mcast_grp_map=(val, prev_val=nil)
-    #  debug "val is of class #{val.class} and is #{val} prev is #{prev_val}"
-    #  # When prev_val is nil, HashDiff doesn't do a `+' on each element, so
-    #  # this
-    #  debug "value of mac_dist = #{@mac_dist_proto}"
-    #  param_hash = {}
-    #  # supress_arp = (@mac_dist_proto == :evpn) ? "supress-arp" : ""
-    #  if prev_val.nil?
-    #    val.each do |fresh_vni, fresh_mgrp|
-    #      param_hash = { name: @name, no_cmd: '', vni: fresh_vni,
-    #                     mcast_grp: fresh_mgrp }
-    #      config_set('vxlan_vtep', 'member_vni_mgrp', param_hash)
-    #    end
-    #    return
-    #  end
-    #  require 'hashdiff'
-    #  hash_diff = HashDiff.diff(prev_val, val)
-    #  debug "hsh diff ; #{hash_diff}"
-    #  return if hash_diff == []
-    #  hash_diff.each do |diff|
-    #    case diff[0]
-    #    when /\+/
-    #      param_hash = { name: @name, no_cmd: '', vni: diff[1],
-    #                     mcast_grp: diff[2] }
-    #    when /\-/
-    #      param_hash = { name: @name, no_cmd: 'no', vni: diff[1],
-    #                     mcast_grp: diff[2] }
-    #    when /~/
-    #      param_hash = { name: @name, no_cmd: 'no', vni: diff[1],
-    #                     mcast_grp: diff[2] }
-    #      config_set('vxlan_vtep', 'member_vni_mgrp', param_hash)
-    #      param_hash = { name: @name, no_cmd: '', vni: diff[1],
-    #                     mcast_grp: diff[3] }
-    #    end
-    #    config_set('vxlan_vtep', 'member_vni_mgrp', param_hash)
-    #  end
-    # rescue CliError => e
-    #  raise "[vxlan_vtep #{@name}] '#{e.command}' : #{e.clierror}"
-    # end
-
-    def mac_distribution
-      mac_dist = config_get('vxlan_vtep', 'mac_distribution', name: @name)
-      if mac_dist.nil?
-        @mac_dist_proto = :flood
-      else
-        @mac_dist_proto = (mac_dist == 'bgp') ? :evpn : :flood
-      end
-      @mac_dist_proto.to_s
-    end
-
-    def mac_distribution=(val)
-      if val == :flood
-        if @mac_dist_proto == :evpn
-          config_set('vxlan_vtep', 'mac_distribution',
-                     name: @name, state: 'no', proto: 'bgp')
-        end
-        @mac_dist_proto = :flood
-      elsif val == :evpn
-        config_set('vxlan_vtep', 'mac_distribution',
-                   name: @name, state: '', proto: 'bgp')
-        @mac_dist_proto = :evpn
+    def host_reachability
+      hr = config_get('vxlan_vtep', 'host_reachability', name: @name)
+      hr == 'bgp' ? 'evpn' : hr
+    end
+
+    def host_reachability=(val)
+      set_args = { name: @name, proto: 'bgp' }
+      if val.to_s == 'flood' && host_reachability == 'evpn'
+        set_args[:state] = 'no'
+      elsif val.to_s == 'evpn'
+        set_args[:state] = ''
       end
```

Comment:
If neither if condition is true then `:state` will not be set when you call config_set. Is that okay?